### PR TITLE
post: ng 5.2

### PR DESCRIPTION
**Pull request by cexbrayat**
WIP (you don't have to review now)

- [x] changelog URL
- [x] `$any()` in templates
- [x] `*ngIf` type guards for `strictNullChecks` and `fullTemplateTypeCheck`
- [x] router params inheritance
- [x] project Ivy

### CODE REVIEW

File: _posts/2017-04-28-what-is-new-angular-4.1.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ Then, with the `strictNullChecks` option activated,
 the compiler will help you when you use this model.
 For example:
 
-   this.user.surname.toLowerCase();
+    this.user.surname.toLowerCase();
```
Comment (cexbrayat) [inline _posts/2017-04-28-what-is-new-angular-4.1.md]: Was missing a space to format this as code

**Comment by cexbrayat** [comment]
@jnizet @clacote Up for review please!
If everything is OK, you can merge and tweet tomorrow morning.
If small typos, you can fix directly.
If bigger questions, I'll try to fix before tomorrow 🙏

File: _posts/2018-01-11-what-is-new-angular-5.2.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    export class PonyComponent {
+      ponyModel: PonyModel;
+      // ponyModel has no field age, so the template should not compile
+    }
+    {% endraw %}
+
+As for `any` in TypeScript, I'm not really fan of using this:
+I usually prefer to have a correct type instead of "cheating" with `any` or `$any()`.
+So this is not really for the day to day use.
+This is more targeted to help the applications using `fullTemplateTypeCheck`
```
Comment (clacote) [inline _posts/2018-01-11-what-is-new-angular-5.2.md]: I had written `targetted` for ever, but it seems you're right, it's `targeted`: http://www.future-perfect.co.uk/grammar-tip/is-it-targetted-or-targeted/

File: _posts/2018-01-11-what-is-new-angular-5.2.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+Angular&nbsp;5.0 introduced the option `fullTemplateTypeCheck`,
+but until now it was not really working.
+I usually don't talk about bug fixes in these release blog posts,
+but for once I think it's interesting: the issues related to this compiler option
+have been fixed (using the `$any()` feature)!
+So you can now try the option to see if it discovers some errors in your templates! ✊
+
+Warning: it can also discover errors in some of your third party libraries.
+For example, we use `ng-bootstrap` and there was two errors in `1.0.0-beta.7`.
+We fixed them, so if you use `ng-bootstrap@1.0.0-beta.8` or a more recent one, you should be OK!
```
Comment (clacote) [inline _posts/2018-01-11-what-is-new-angular-5.2.md]: A link to you commit/PR would contribute to the fame

File: _posts/2018-01-11-what-is-new-angular-5.2.md (hunk 2 above)
Comment (jnizet) [inline _posts/2018-01-11-what-is-new-angular-5.2.md]: This is rather intended to help?

File: _posts/2018-01-11-what-is-new-angular-5.2.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+As for `any` in TypeScript, I'm not really fan of using this:
+I usually prefer to have a correct type instead of "cheating" with `any` or `$any()`.
+So this is not really for the day to day use.
+This is more targeted to help the applications using `fullTemplateTypeCheck`
+which can raise type errors hard to fix, usually from third party libraries.
+This was also introduced for internal use in the framework (see below).
+
+## Compiler
+
+Angular&nbsp;5.0 introduced the option `fullTemplateTypeCheck`,
```
Comment (jnizet) [inline _posts/2018-01-11-what-is-new-angular-5.2.md]: Bizarre, you talk about this in two different sections, to basically say the ame thing.

File: _posts/2018-01-11-what-is-new-angular-5.2.md (hunk 2 above)
Comment (cexbrayat) [reply _posts/2018-01-11-what-is-new-angular-5.2.md]: Ok to change

File: _posts/2018-01-11-what-is-new-angular-5.2.md (hunk 4 above)
Comment (cexbrayat) [reply _posts/2018-01-11-what-is-new-angular-5.2.md]: Ok to change

**Comment by cexbrayat** [comment]
I let you fix and publish if you can

**Comment by clacote** [comment]
@jnizet can you deal with the fix?

File: _posts/2018-01-11-what-is-new-angular-5.2.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+but until now it was not really working.
```
Comment (acrepet) [inline _posts/2018-01-11-what-is-new-angular-5.2.md]: -> "but until now it has not been working " present perfect continuous should be used here, nan?

File: _posts/2018-01-11-what-is-new-angular-5.2.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+I usually don't talk about bug fixes in these release blog posts,
+but for once I think it's interesting: the issues related to this compiler option
+have been fixed (using the `$any()` feature)!
+So you can now try the option to see if it discovers some errors in your templates! ✊
+
+Warning: it can also discover errors in some of your third party libraries.
+For example, we use `ng-bootstrap` and there was two errors in `1.0.0-beta.7`.
```
Comment (acrepet) [inline _posts/2018-01-11-what-is-new-angular-5.2.md]: -> there were two errors

**Comment by jnizet** [comment]
I'll do it.

File: _posts/2018-01-11-what-is-new-angular-5.2.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+We can now simply write:
+
+    {% raw %}
+    {{ ponyModel.name }}
+    {% endraw %}
+
+and the compiler will understand the situation!
+
+Note that this a generic feature: if you wrote your own structural directive,
+that works like an `*ngIf`, you can also leverage this type guard feature by adding
```
Comment (acrepet) [inline _posts/2018-01-11-what-is-new-angular-5.2.md]: with the past "wrote" we should use "should", nan?
-> "if you wrote your own structural directive, that should work" like..."

**Comment by jnizet** [comment]
Deployed. @clacote I let you tweet about it.

**Comment by clacote** [comment]
💋

https://twitter.com/NinjaSquad/status/951797459864424450

**Pull request merged**